src/reducetensor.cpp

```diff
@@ -43,9 +43,10 @@
 #include <iostream>
 #include <sstream>
 
-MIOPEN_DECLARE_ENV_VAR(MIOPEN_DISABLE_DYNAMIC_REDUCTION);
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_DYNAMIC_REDUCTION);
 
 #define WORKAROUND_MIOPEN_ISSUE_557 1
+#define WORKAROUND_ISSUE_1123 ((HIP_PACKAGE_VERSION_FLAT == 4003000000ULL) && MIOPEN_BACKEND_HIP)
```

Comment:
This is wrong. For 4.3 HIP version is 4.3.21300, For 4.3.1 it is 4.3.21331 (according to CI logs). This must be changed to
```C++
#define WORKAROUND_ISSUE_1123 (HIP_PACKAGE_VERSION_FLAT >= 4003000000ULL && HIP_PACKAGE_VERSION_FLAT <= 4003021300ULL && MIOPEN_BACKEND_HIP)
```
Otherwise dynamic reduction will NOT be automatically enabled for 4.3.1.

Reply:
@junliume My bad; I've forgot the details of the HIP release numbering scheme.

Reply:
Missed it during review.

Reply:
RIGHT NOW WORKAROUND IS NEVER ENABLED.

Reply:
@atamazov No problem! Testing in this draft now: #1126
Will create separate PRs for the double purposed draft.

Reply:
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/ci_gfx1030_ocl_to_hip/3/pipeline/11

> [2021-08-30T20:31:02.411Z] 79/82 Test #80: test_reduce_double .........................   Passed  880.12 sec

This particular test is still okay but the other tests were terminated somehow. I'll keep monitoring.